base_wamas_ubl/lib/wamas/wamas2wamas.py

```diff
@@ -0,0 +1,203 @@
+#!/usr/bin/python3
+
+import getopt
+import logging
+import re
+import sys
+from collections import OrderedDict
+
+from ubl2wamas import (  # noqa: F401
+    DICT_WAMAS_GRAMMAR,
+    _get_current_datetime,
+    _get_sequence_number,
+    set_value_to_string,
+)
+from utils import *  # noqa: F403
+from wamas2ubl import fw2dict, telegram_header_grammar
+from wamas_grammar import (  # noqa: F401
+    ausk,
+    auskq,
+    ausp,
+    kretk,
+    kretkq,
+    kretp,
+    kretpq,
+    watekq,
+    watepq,
+    weak,
+    weakq,
+    weap,
+    weapq,
+)
+
+_logger = logging.getLogger("wamas2wamas")
+
+
+LST_VALID_TELEGRAM_IN = [
+    "AUSK",
+    "AUSP",
+    "KRETK",
+    "KRETP",
+    "WATEK",
+    "WATEP",
+    "WEAK",
+    "WEAP",
+]
+
+
+DICT_CONVERT_WAMAS_TYPE = {
+    "AUSK": ["AUSKQ", "WATEKQ"], 
+    "AUSP": ["WATEPQ"],
+    "AUSKQ": ["AUSK"],
+    "KRETK": ["KRETKQ"],
+    "KRETP": ["KRETPQ"],
+    "WEAK": ["WEAKQ"],
+    "WEAP": ["WEAPQ"],
+}
+
+
+DICT_WAMAS_GRAMMAR_OUT = {
+    "auskq": auskq.grammar_convert,
+    "kretkq": kretkq.grammar_convert,
+    "kretpq": kretpq.grammar_convert,
+    "watekq": watekq.grammar_convert,
+    "watepq": watepq.grammar_convert,
+    "weakq": weakq.grammar_convert,
+    "weapq": weapq.grammar_convert,
+}
+
+
+DICT_PARENT_KEY = {
+    "WATEKQ": ["IvTek_TeId"]
+}
+
+
+DICT_CHILD_KEY = {
+    "WATEPQ": {
+        "IvTep_TeId": "IvTek_TeId"
+    }
+}
+
+
+def get_simple_grammar(telegram_type):
+    grammar = DICT_WAMAS_GRAMMAR[telegram_type.lower()]
+    simple_grammar = OrderedDict()
+    for field in grammar:
+        if field in telegram_header_grammar.keys():
+            continue
+        simple_grammar[field] = grammar[field]["length"]
+    return simple_grammar
+
+
+def _process_wamas_in(infile):
```

Comment:
very similar to wamas2dict() no? can't be factored somewhere?

Reply:
hi @nilshamerlinck I understand what you mean, so, because we have 3 branches:
- `16.0-add-base_wamas`: the source branch
- `16.0-add-base_wamas_ubl-wamas_generator`: the refactoring branch
- `16.0-add-base_wamas_ubl-wamas2wamas`: this one - wamas to wamas

--> So I decide:
- this PR will be closed
- `16.0-add-base_wamas_ubl-wamas2wamas` will depend on `16.0-add-base_wamas_ubl-wamas_generator`
- refactoring `_process_wamas_in` & `wamas2dict` and move it to `utils.py`
- the new PR of `wamas2wamas` will be created later after refactoring